.github/workflows/update-website.yml

```diff
@@ -0,0 +1,41 @@
+name: Update Website
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'docs/src/**'
+
+jobs:
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    container: asterinas/asterinas:0.4.0
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'asterinas/asterinas'
+          path: 'asterinas'
+      
+      - name: Build the website
+        run: |
+          cd asterinas/docs
+          mdbook build
+
+      - name: Deploy website
+        env:
+          DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
+        run: |
+          echo "$DEPLOY_KEY " > ./deploy_key
+          chmod 600 ./deploy_key
+          ssh-keyscan -t rsa github.com >> ./known_hosts
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "github-actions[bot]"
+          GIT_SSH_COMMAND="ssh -i ./deploy_key -o UserKnownHostsFile=./known_hosts" git clone git@github.com:asterinas/book.git
+          cd book
+          git checkout main
+          cp -r ../asterinas/docs/book/* .
+          git add .
+          git commit -m "Update website"
```

Comment:
Adding a new commit is ok but not ideal. First, keeping the entire history of the generated content of MdBook is of no use.  The source of the Book is version controlled by the `asterinas/asterinas` repo. That's what matters. The latest content of `asterinas/book` matters, but its past versions are not. Second, the command of `cp -r ../asterinas/docs/book/* .` can only add new files or updates existing files, but not delete old files.

Reply:
I understand the point of this, I will try to modify the workflow to accomplish this.